mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2442,7 +2480,7 @@ public LoanTransaction handlePayDisbursementTransaction(final Long chargeId, fin
             if (this.loanProduct.isMultiDisburseLoan()) {
                 for (final LoanDisbursementDetails details : this.disbursementDetails) {
                     details.updateActualDisbursementDate(null);
-                    details.resetPrincipal();
+                    details.updatePrincipal(this.approvedPrincipal);
```

Comment:
why this is required? we shouldn't change any amounts on undodisbursal or undoapproval